x/interchainstaking/keeper/delegation_test.go

```diff
@@ -611,65 +613,298 @@ func (s *KeeperTestSuite) TestStoreGetDeleteDelegation() {
+			setStatements: func(ctx sdk.Context, quicksilver *app.Quicksilver) {
+				cutOffTime := ctx.BlockTime().AddDate(0, 0, -1)
+				pendingRecieptTime := cutOffTime.Add(-2 * time.Hour)
+				excludedReciptTime := cutOffTime.Add(-3 * time.Hour)
+
+				rcpt1 := types.Receipt{
+					ChainId: s.chainB.ChainID,
+					Sender:  userAddress,
+					Txhash:  "TestDeposit01",
+					Amount: sdk.NewCoins(
+						sdk.NewCoin(
+							denom,
+							sdk.NewIntFromUint64(2000000), // 20% deposit
+						),
+					),
+					FirstSeen: &pendingRecieptTime,
+					Completed: nil,
+				}
+
+				rcpt2 := types.Receipt{
+					ChainId: s.chainB.ChainID,
+					Sender:  userAddress,
+					Txhash:  "TestDeposit02",
+					Amount: sdk.NewCoins(
+						sdk.NewCoin(
+							denom,
+							sdk.NewIntFromUint64(100), // 20% deposit
+						),
+					),
+					FirstSeen: &excludedReciptTime,
+					Completed: nil,
+				}
+				quicksilver.InterchainstakingKeeper.SetReceipt(ctx, rcpt1)
+				quicksilver.InterchainstakingKeeper.SetReceipt(ctx, rcpt2)
+			},
+			delAddrBalance: sdk.NewCoin("uatom", sdkmath.NewInt(2000100)),
+			assertStatements: func(ctx sdk.Context, quicksilver *app.Quicksilver) bool {
+				count := 0
+				zone, found := quicksilver.InterchainstakingKeeper.GetZone(ctx, s.chainB.ChainID)
+				s.Require().True(found)
+				quicksilver.InterchainstakingKeeper.IterateZoneReceipts(ctx, &zone, func(index int64, receiptInfo types.Receipt) (stop bool) {
+					if receiptInfo.Completed == nil {
+						count++
+					}
+					return false
+				})
+				s.Require().Equal(0, count)
+				return true
+			},
+			mockAck:            true,
+			expectedDelegation: sdk.NewCoins(sdk.NewCoin("uatom", sdk.NewInt(2000100))),
+		},
+	}
 
-		app.InterchainstakingKeeper.SetReceipt(ctx, receipt)
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			s.SetupTest()
+			s.setupTestZones()
 
-		zone, found := app.InterchainstakingKeeper.GetZone(ctx, s.chainB.ChainID)
-		s.Require().True(found)
+			quicksilver := s.GetQuicksilverApp(s.chainA)
+			ctx := s.chainA.GetContext()
 
-		count, err := app.InterchainstakingKeeper.FlushOutstandingDelegations(ctx, &zone)
-		s.Require().NoError(err)
-		s.Require().Equal(1, count)
-	})
+			test.setStatements(ctx, quicksilver)
+			zone, found := quicksilver.InterchainstakingKeeper.GetZone(ctx, s.chainB.ChainID)
+			s.Require().True(found)
+			before := quicksilver.InterchainstakingKeeper.AllReceipts(ctx)
+			err := quicksilver.InterchainstakingKeeper.FlushOutstandingDelegations(ctx, &zone, test.delAddrBalance)
+			if test.mockAck {
+				var msgs []sdk.Msg
+				allocations, err := quicksilver.InterchainstakingKeeper.DeterminePlanForDelegation(ctx, &zone, test.expectedDelegation)
+				s.Require().NoError(err)
+				msgs = append(msgs, quicksilver.InterchainstakingKeeper.PrepareDelegationMessagesForCoins(&zone, allocations)...)
+				for _, msg := range msgs {
+					err := quicksilver.InterchainstakingKeeper.HandleDelegate(ctx, msg, "batch/1577836910")
```

Comment:
Will this work? The times against the receipts are dynamic w.r.t test run time, and this batch time is fixed.

Reply:
actually its not dynamic the block time is 1st Jan 2020